sunpy/coordinates/metaframes.py

```diff
@@ -178,13 +179,27 @@ def __new__(cls, *args, **kwargs):
 
     def __init__(self, *args, **kwargs):
         # Validate inputs
-        if kwargs['base'].obstime is None:
+        base_frame = kwargs.get('base')
+        if base_frame is None or base_frame.obstime is None:
             raise ValueError("The base coordinate frame must have a defined `obstime`.")
 
-        if 'rotated_time' in kwargs:
-            rotated_time = parse_time(kwargs['rotated_time'])
-            kwargs['duration'] = (rotated_time - kwargs['base'].obstime).to('day')
-            kwargs.pop('rotated_time')
+        duration = kwargs.pop('duration', None)
+        rotated_time = kwargs.pop('rotated_time', None)
+
+        if duration is not None and rotated_time is not None:
+            raise ValueError("Specify either `duration` or `rotated_time`, not both.")
+
+        if not isinstance(duration, TimeDelta) and not isinstance(duration, u.Quantity) and duration is not None:
+            raise ValueError("`duration` must be a `TimeDelta` or `Quantity` object.")
+
```

Comment:
```suggestion
```
We shouldn't require the type.  All that matters is that `duration` can add to `base_frame.obstime`.

Reply:
Alright.